### ISSUE TITLE
How to resize area cropped to square in rectangle

### ISSUE THREAD

**Issue by Bitmaudit** [denis-kalinichenko/js-cropper#1]
Hi,
thanks for your code, I've a question for you, this is my initialize string:

cropper_bg = new Cropper({width:350,height:90,onInit: function (crop) {crop.setWidth(600);crop.setHeight(200),crop.setZoom(0.1);},onChange: function(crop) {}});cropper_bg.reset();

the area was just correct but the layer trasparen of crop was the same, always squre, can you help me?

Thanks!

**Comment by vivek-pai** [comment]
Yes, I have the same problem. How do you change the actual crop size?

**Comment by Bitmaudit** [comment]
i've used alternative script

js-cropper-master.zip

$html='<script type="text/javascript">';
$html.='$(\'#id_ref_upload_avatar\').html(\'<img id="id_ref_cropper_avatar"
src="'.$filelink.'">\');';
$html.='var image=document.getElementById(\'id_ref_cropper_avatar\');';
$html.=' var options =
{
aspectRatio:10/10,
viewMode:3,
dragMode:\'move\',
toggleDragModeOnDblclick:true,
cropBoxMovable:true,
cropBoxResizable:false,
guides:false,
zoomOnWheel:false,
movable:true,
scalable:false,
ready: function(e)
{
var imageData = cropper_avatar.getImageData();
var containerData = cropper_avatar.getContainerData();
//Image original dimension
var image_data_natural_width = imageData.naturalWidth;
var image_data_natural_heigh = imageData.naturalHeight;
//Image crop dimension
var image_data_width = imageData.width;
var image_data_height = imageData.height;
//Container cropper_avatar dimension
var container_width=containerData.width;
var container_height=containerData.height;
var
zoom_min_value=(image_data_natural_width/image_data_width)/image_data_width;

$(\'#zoom_slider_avatar\').val(0);
$(\'#min_zoom_avatar\').val(zoom_min_value);
cropper_avatar.zoomTo(zoom_min_value);
}};';
$html.='Cropper.setDefaults(options);';
$html.='var cropper_avatar = new Cropper(image,{});';
$html.='$(\'#id_cropper_button_avatar\').css(\'display\',\'block\');';
$html.='$(\'#status_upload_avatar\').html(\'\');';
$html.='$(\'#id_tmp_file_avatar\').val(\''.$filelink.'\');';
$html.='</script><input type="hidden" name="filelink_return_avatar"
id="filelink_return_avatar" value="'.$filelink.'" />';


Il giorno sab 11 lug 2020 alle ore 06:11 Vivek Pai <notifications@github.com>
ha scritto:

**Comment by Bitmaudit** [comment]
change aspect ratio in aspectRatio:12/3 or what you want
